tests/integration/cmd_app_test.go

```diff
@@ -61,14 +63,14 @@ var _ = Describe("odo app command tests", func() {
 			appListOutput := helper.CmdShouldPass("odo", "app", "list", "--project", commonVar.Project)
 			Expect(appListOutput).To(ContainSubstring(appName))
 			actualCompListJSON := helper.CmdShouldPass("odo", "list", "-o", "json", "--project", commonVar.Project)
-
-			desiredCompListJSON := fmt.Sprintf(`{"kind":"List","apiVersion":"odo.dev/v1alpha1","metadata":{},"s2iComponents":[{"kind":"Component","apiVersion":"odo.dev/v1alpha1","metadata":{"name":"nodejs","namespace":"%s","creationTimestamp":null},"spec":{"app":"app","type":"nodejs","sourceType":"local","env":[{"name":"DEBUG_PORT","value":"5858"}]},"status":{"state":"Pushed"}}],"devfileComponents":[]}`, commonVar.Project)
-			Expect(desiredCompListJSON).Should(MatchJSON(actualCompListJSON))
-
+			valuesL := gjson.GetMany(actualCompListJSON, "kind", "s2iComponents.0.metadata.name", "s2iComponents.0.metadata.namespace")
+			expectedL := []string{"List", "nodejs", commonVar.Project}
+			Expect(helper.GjsonMatcher(valuesL, expectedL)).To(Equal(true))
 			helper.CmdShouldPass("odo", "app", "describe", "--project", commonVar.Project)
-			desiredDesAppJSON := fmt.Sprintf(`{"kind":"Application","apiVersion":"odo.dev/v1alpha1","metadata":{"name":"app","namespace":"%s","creationTimestamp":null},"spec":{"components": ["nodejs"]}}`, commonVar.Project)
 			actualDesAppJSON := helper.CmdShouldPass("odo", "app", "describe", "app", "-o", "json", "--project", commonVar.Project)
-			Expect(desiredDesAppJSON).Should(MatchJSON(actualDesAppJSON))
```

Comment:
I think we don't need `MatchJSON` function anymore. Could you please remove this function from library as well.

Reply:
MatchJSON is a function provided by GOMEGA, so I will leave it there